### ISSUE TITLE
Can't update category via REST

### ISSUE THREAD

**Issue by marcelschmidt1337** [fcturner/passwords#203]
When I try PUT on an existing category I'm getting an internal server error 500.
I'm not quite sure if this is supported (didn't find anything update related in categoryservice.php) but would be nice to have this option.

My test configuration:
OC 9.0.2
Apache 2.4
MySQL + PHP 5.6.2

**Comment by fcturner** [comment]
Should work, I'll look into this.

**Comment by fcturner** [comment]
Fixed it for SQLite, it contained an old error. 
So if you are using SQLite, please try again with latest master.

**Comment by intirix** [comment]
This is the error I get with the latest v19 branch against MySQL:

{"reqId":"ufusb9fTGfbnHg9GHU5B","remoteAddr":"172.17.0.1","app":"index","message":"Exception: {\"Exception\":\"ReflectionException\",
\"Message\":\"Method OCA\\\\Passwords\\\\Controller\\\\CategoryApiController::update() does not exist\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/ocdev\\\/lib\\\/private\\\/appframework\\\/utility\\\/controllermethodreflector.php(53): ReflectionMethod->__construct(Object(OCA\\\\Passwords\\\\Controller\\\\CategoryApiController), 'update')\\n#1 \\\/var\\\/www\\\/html\\\/ocdev\\\/lib\\\/private\\\/appframework\\\/http\\\/dispatcher.php(85): OC\\\\AppFramework\\\\Utility\\\\ControllerMethodReflector->reflect(Object(OCA\\
\\Passwords\\\\Controller\\\\CategoryApiController), 'update')\\n#2 \\\/var\\\/www\\\/html\\\/ocdev\\\/lib\\\/private\\\/appframework
\\\/app.php(110): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OCA\\\\Passwords\\\\Controller\\\\CategoryApiController), 'update')\\n#3 \\\/var\\\/www\\\/html\\\/ocdev\\\/lib\\\/private\\\/appframework\\\/routing\\\/routeactionhandler.php(45): OC\\\\AppFramework\\\\App::main('CategoryApiCont...', 'update', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#4 
[internal function]: OC\\\\AppFramework\\\\routing\\\\RouteActionHandler->__invoke(Array)\\n#5 \\\/var\\\/www\\\/html\\\/ocdev\\\/lib\\\/private\\\/route\\\/router.php(276): call_user_func(Object(OC\\\\AppFramework\\\\routing\\\\RouteActionHandler), Array)\\n#6 \\\/
var\\\/www\\\/html\\\/ocdev\\\/lib\\\/base.php(882): OC\\\\Route\\\\Router->match('\\\/apps\\\/passwords...')\\n#7 \\\/var\\\/www\\\/html\\\/ocdev\\\/index.php(39): OC::handleRequest()\\n#8 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/ocdev\\\/lib\\\/private\\\/appframework\\\/utility\\\/controllermethodreflector.php\",\"Line\":53}","level":3,"time":"2016-06-17T17:43:17+00:00","method":"PUT","url":"\/ocdev\/index.php\/apps\/passwords\/api\/0.1\/categories\/1","user":"pwuser1"}


categoryapicontroller.php does not have an update() method

**Comment by fcturner** [comment]
But UPDATE isn't a valid value for a RESTful API, updates are done using PATCH. Does that help? I don't know much about API's, just implemented it using a pre made app from ownCloud.

**Comment by intirix** [comment]
I will play around with PATCH.  That method is currently blocked by my webserver.  The REST API documentation in the wiki says to use POST to update a resource.

https://github.com/fcturner/passwords/wiki/ownCloud-Passwords-%7C-RESTful-API

**Comment by fcturner** [comment]
How annoyingly stupid of me. I ditched `PATCH` due to #148.
ownCloud doesn't support it, as far as I can tell. Indeed, `PUT` en `POST` are the way to go.

**Comment by fcturner** [comment]
Is this still relevant? Is there anything I can do?
Please reopen issue if a problem still exists.
